Package/Runtime/CrashKonijn.Goap.Runtime/Sensors/LocalTargetSensorBase.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using CrashKonijn.Agent.Core;
+using CrashKonijn.Goap.Core;
+
+namespace CrashKonijn.Goap.Runtime
+{
+    public abstract class LocalTargetSensorBase : ILocalTargetSensor
+    {
+        public ITargetKey Key => this.Config.Key;
+        public ITargetSensorConfig Config { get; private set; }
+        public void SetConfig(ITargetSensorConfig config) => this.Config = config;
+        
+        public abstract void Created();
+        public abstract void Update();
+        public Type[] GetKeys() => new[] { this.Key.GetType() };
+
+        public void Sense(IWorldData worldData, IActionReceiver agent, IComponentReference references)
+        {
+            worldData.SetTarget(this.Key, this.Sense(agent, references));
+        }
+
+        public virtual ITarget Sense(IActionReceiver agent, IComponentReference references)
+        {
+#pragma warning disable CS0618 // Type or member is obsolete
+            return this.Sense(agent as IMonoAgent, references);
```

Comment:
This is a passthru for now? Forcing folks to use an IActionReceiver but internally its the same for now?

Reply:
So, theory:

In v2 the `ActionReceiver` was always the `AgentBehaviour`. In the past I've discussed being able to layer multiple algorithems on top of eachother. For example let a GOAP decide how to walk and let another GOAP decide what to do, or even let a GOAP decide the goal it should do.

When taking this layered approach not every `IActionReceiver` is necessarily the `AgentBehaviour` and that's why I've split them up.

The method with the `IActionReceiver` is the correct method, the other one is obsolete. I've re-added this obsolete method yesterday, because now their sensors won't have compile issues but show a warning instead.

After a certain amount of time the obsolete method will be removed again